qiskit_experiments/randomized_benchmarking/interleaved_rb_analysis.py

```diff
@@ -0,0 +1,163 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+"""
+Interleaved RB analysis class.
+"""
+from typing import Optional, List
+import numpy as np
+from qiskit_experiments.analysis.curve_fitting import (
+    process_multi_curve_data,
+    multi_curve_fit,
+)
+
+from qiskit_experiments.analysis.plotting import plot_curve_fit, plot_scatter, plot_errorbar
+from qiskit_experiments.analysis.data_processing import (
+    level2_probability,
+    mean_xy_data,
+)
+from .rb_analysis import RBAnalysis
+
+try:
+    from matplotlib import pyplot as plt
+
+    HAS_MATPLOTLIB = True
+except ImportError:
+    HAS_MATPLOTLIB = False
+
+
+class InterleavedRBAnalysis(RBAnalysis):
+    """Interleaved RB Analysis class."""
+
+    # pylint: disable=invalid-name, unused-variable, unused-argument, attribute-defined-outside-init
```

Comment:
Do you really need to disable that much?

Reply:
`invalid-name` allows us to use the names from the mathematical formulation.
`unused-argument` is required to allow fit functions which don't use all parameters (e.g.the int RB function doesn't use the std RB alpha)
`attribute-defined-outside-init` is required if we want to avoid passing the number of qubits to all relevant functions - I can change it.

Have to admit I'm not sure why `unused-variable` was needed (something similar to `unused-argument`, I think)

Reply:
The attribute-defined-outside-init should not be disabled. I put a fix for this for standard RB fitter in #38. I'm skeptical on the unused variable/arguments too. There probably shouldn't be any need to disable them and they might be showing an issue with the code.

Reply:
The standard Python solution is using _ to avoid naming the parameters. I don't like this approach in this case since it hides information which can help the reader, but I've used it anyway to avoid disabling the linter.